servers/src/main/java/tachyon/master/permission/FileSystemPermissionChecker.java

```diff
@@ -58,18 +109,80 @@ public static synchronized void initializeFileSystem(boolean permissionCheckEnab
    * @param user who requests access permission
    * @param groups in which user belongs to
    * @param action requested {@link FileSystemAction} by user
-   * @param inodes all the inodes retrieved by traversing the path
-   * @throws IOException if permission checking fails
+   * @param path the path to check permission on
+   * @param fileInfos file infos of all the inodes retrieved by traversing the path
+   * @throws AccessControlException if permission checking fails
    */
-  public static void checkPermission(User user, List<String> groups, FileSystemAction action,
-      List<Inode> inodes) throws IOException {
+  private static void checkByFileInfos(String user, List<String> groups, FileSystemAction action,
+      String path, List<FileInfo> fileInfos) throws AccessControlException {
+    int size = fileInfos.size();
+    assert size > 0;
```

Comment:
please use 
`Preconditions.checkArgument(size > 0)` 
rather than assert. In Tachyon we don't use `assert` in general.